### ISSUE TITLE
Swift 4 update

### ISSUE THREAD

**Issue by artemkalinovsky** [RLovelett/langserver-swift#29]
Hi @RLovelett ,

Your Swift extension for Visual Studio Code is awesome and I'm using it everyday with Xcode.
Do you plan to continue support and release update for Swift 4?

Thanks

**Comment by TofPlay** [comment]
Same question as @artemkalinovsky

**Comment by RLovelett** [comment]
I have every intention of doing so. In fact, I've started the process of upgrading to Swift 4. I mostly have it compiling against Swift 4 though I haven't fully tested everything.

The first challenge is that the [API for the SwiftPM integration](https://github.com/RLovelett/langserver-swift/blob/ddac660fb8355039ac5625ecf6c1d6787d95e255/Package.swift#L16) is drastically different between 3 and 4.

The second challenge is that I live in Florida and we were affected by Irma. We went over a week without water, power or internet. Things are starting to return to normal but I've spent the last few weekends (which is when I typically work on this) fixing broken things at the house and clearing debris from my yard and family member's yards.

Hopefully soon I will be able to get back to a more typical work cycle.
